puppet/lib/puppet/reports/puppetdb.rb

```diff
@@ -73,6 +75,42 @@ def build_events_list
     end
   end
 
+  # @return Array[Hash]
+  # @api private
+  def build_logs_list
+    profile("Build logs list (count: #{logs.count})",
+            [:puppetdb, :logs_list, :build]) do
+      log_list = []
+      logs.each do |log|
+        log_hash = {
+          'file' => log.file,
+          'line' => log.line,
+          'level' => log.level,
+          'message' => log.message,
+          'source' => log.source,
+          'tags' => [*log.tags],
+          'time' => Puppet::Util::Puppetdb.to_wire_time(log.time),
+        }
+        log_list.push(log_hash)
+      end
+
+      log_list
+    end
+  end
+  def build_metrics_list
+    profile("Build metrics list (count: #{metrics.count})",
+            [:puppetdb, :metrics_list, :build]) do
+      metrics_list = []
+      metrics.each do |metric_kv|
```

Comment:
It's a key-value pair like this:

["time",
 #<Puppet::Util::Metric:0x007ff51a1198a8
  @label="Time",
  @name="time",
  @values=
   [["schedule", "Schedule", 0.000283],
    ["notify", "Notify", 0.00052],
    ["filebucket", "Filebucket", 7.2e-05],
    ["config_retrieval", "Config retrieval", 0.318574],
    ["total", "Total", 0.31944900000000004]]>]

where the element being selected is the second, not the first. I don't have an issue with naming it that way, but I you have an alternative I'm happy to change it.